tests/utilities/discovery_service_process.py

```diff
@@ -0,0 +1,38 @@
+"""Generator class to create and terminate Discovery Service instance."""
```

Comment:
This is a context manager, not a generator.

```suggestion
"""Class to create and terminate Discovery Service instance."""
```